### ISSUE TITLE
Response Wrapper

### ISSUE THREAD

**Issue by pfulop** [apollostack/apollo-client#284]
Is there way to somehow wrap responses from graphql server? I want to know if server returned 401 or refreshed token.

**Comment by stubailo** [comment]
Can you explain a bit more what you mean by this? For example, a hypothetical code snippet of what you would like to do? I'm not quite sure what wrapping a response would mean.

**Comment by pfulop** [comment]
For instance your network interface allows to applyMiddleware to work with request object. It would be nice if I could apply middleware to response, and have something like general error check.

`networkInterface.use([{
  applyMiddleware(res, next) {
    if (!res.status===401) {
      dispatch(logout());
    }
    next();
  }
}]);`

Therefore I wouldn't need to write same code for every query and mutation response.

**Comment by monteirocode** [comment]
Yes! 🙌  I was just searching the docs for a way to to this, maybe we could call it an `afterware `?

```js
networkInterface.use([{
  applyAfterware(response, next) {

    if (response.status === 401) {
      logout()
    }

    next()
  }
}])
```

**Comment by stubailo** [comment]
Sure, would you be interested in submitting a PR for this?

**Comment by monteirocode** [comment]
@stubailo I'm a very junior developer but I can give it a try

**Comment by drager** [comment]
This can be closed now since #399 is merged?

**Comment by Poincare** [comment]
Yup, this can be closed.
